lib/cretonne/src/ir/funcname.rs

```diff
@@ -14,65 +13,30 @@ use std::ascii::AsciiExt;
 pub struct FunctionName(String);
 
 impl FunctionName {
-    /// Create new function name equal to `s`.
-    pub fn new<S: Into<String>>(s: S) -> FunctionName {
+    /// Creates a new function name from a string.
+    ///
+    /// Caller should validate that the string contains only
+    /// ASCII alphanumerical characters and `_`.
+    pub fn from_string(s: &str) -> FunctionName {
```

Comment:
I think `with_str` would be a better name for this constructor. The`from_string` name implies a *conversion constructor* which would be expected to consume a `String`. See the [Rust naming conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html).

I don't think the comment is correct. The caller should be allowed to pass any string, and it will be printed in `#xxx` form if it is not pure ASCII.